Assets/UI/Popups/greatpeoplepopup.lua

```diff
@@ -146,302 +172,303 @@ function GetBiographyTextTable( individualID:number )
-
-        if (CQUI_isPreferredHeightsComputed == false) then
-            --CQUI_preferredRecruitScrollSize = 1000;  -- for quick testing
-            if (CQUI_preferredRecruitScrollSize > (CQUI_screenHeight / 4)) then
-                CQUI_preferredRecruitScrollSize = (CQUI_screenHeight / 4);
-            end
+        CQUI_isPreferredHeightsComputed = true;
+    end
 
-            -- 670 is the default Instance size in the XML... 86 is a number that represents some undocumented thing
-            CQUI_preferredInstanceSize =  670 - 86 + CQUI_preferredRecruitScrollSize;
-            -- CQUI_preferredInstanceSize = 3000 -- for quick testing
-            if (CQUI_preferredInstanceSize > (CQUI_screenHeight - CQUI_instanceMargin)) then
-                -- The instance area cannot be bigger than the screen height minus CQUI_instanceMargin:
-                -- When the Gold/Faith (or Recruit/Pass) buttons are 5px above the PeopleScroller horizontal scroll, 
-                -- the PanelInstance is CQUI_instanceMargin pixels less in height than the PeopleContainer, which is defined as 768 in the XML.
-                -- These adjustments are necessary to properly fit the screen
-                local prevPreferredInstanceSize = CQUI_preferredInstanceSize;
-                CQUI_preferredInstanceSize = CQUI_screenHeight - CQUI_instanceMargin;
-
-                -- Instead of shrinking the recruit scroll size, instead shrink the EffectsStackScroller, as there's typically room to spare in that section
-                -- 240 is the value defined in the XML.  We cannot do a GetSizeY here because subsequent calls to this function
-                -- would update the smaller value, eventually shrinking the control to less than zero.
-                local effectStackScrollerAdjustment = prevPreferredInstanceSize - CQUI_preferredInstanceSize;
-                CQUI_preferredEffectsScrollSize = 240 - effectStackScrollerAdjustment;
-            end
+    -- CQUI : change size
+    -- CQUI DEBUG TEST :
+    -- CQUI_preferredRecruitScrollSize = 300;
+    instance.Content:SetSizeY(CQUI_preferredInstanceSize);
+    instance.EffectStackScroller:SetSizeY(CQUI_preferredEffectsScrollSize);
+    instance.RecruitScroll:SetSizeY(CQUI_preferredRecruitScrollSize);
+end
 
-            CQUI_isPreferredHeightsComputed = true;
-        end
+-- ===========================================================================
+function ResetGreatPeopleInstances()
+    m_greatPersonPanelIM:ResetInstances();
+    m_greatPersonRowIM:ResetInstances();
+end
 
-        -- CQUI : change size
-        -- CQUI DEBUG TEST :
-        -- CQUI_preferredRecruitScrollSize = 300;
-        instance.Content:SetSizeY(CQUI_preferredInstanceSize);
-        instance.EffectStackScroller:SetSizeY(CQUI_preferredEffectsScrollSize);
-        instance.RecruitScroll:SetSizeY(CQUI_preferredRecruitScrollSize);
+-- ===========================================================================
```

Comment:
ViewCurrent is unchanged, save for moving the huge chunk of code that existed in the _for loop_ at line 466 up to the "AddRecruit" function